### PR TITLE
Fix flaky iOS integration test

### PR DESCRIPTION
**Pull request by denrase**
#skip-changelog

## :scroll: Description

Use latest macos to run integration tests.

Relates to #1448

## :green_heart: How did you test it?

Run on CI

## :pencil: Checklist

- [x] I reviewed submitted code
- [ ] I added tests to verify changes
- [ ] No new PII added or SDK only sends newly added PII if `sendDefaultPii` is enabled
- [ ] I updated the docs if needed
- [x] All tests passing
- [x] No breaking changes

## :crystal_ball: Next steps

We should probably run it a couple of times to verify it always works.

### CODE REVIEW

**Comment by denrase** [comment]
@marandaneto I updated to 'macos-13' and the tests did run now, but I can't say if this was just luck and they remain flaky overall. Build is still slow, alsomst 20 min before the tests were started.

Bildschirmfoto 2023-06-26 um 16 40 10

File: .github/workflows/flutter_integration_test.yml [hunk 1]
```diff
@@ -20,22 +20,56 @@ jobs:
...
+       run: flutter pub get
+
+     - name: launch android emulator & run android integration test
+       uses: reactivecircus/android-emulator-runner@50986b1464923454c95e261820bc626f38490ec0 #pin@v2.27.0
+       with:
+         working-directory: ./flutter/example
+         api-level: 21
+         arch: x86_64
+         profile: Nexus 6
+         script: flutter test integration_test/integration_test.dart --verbose
```
Comment (marandaneto) [inline .github/workflows/flutter_integration_test.yml:56]: We can apply those changes after merged here https://github.com/getsentry/sentry-dart/pull/1543
This helps with speeding up and flakiness.

File: flutter/example/integration_test/integration_test.dart [hunk 2]
```diff
@@ -18,7 +18,6 @@ void main() {
         bundle: SentryAssetBundle(enableStructuredDataTracing: true),
         child: const MyApp(),
       )));
-      await tester.pumpAndSettle();
```
Comment (marandaneto) [inline flutter/example/integration_test/integration_test.dart:21]: Isn't that needed anymore?

Comment (denrase) [reply flutter/example/integration_test/integration_test.dart:21]: Does not make a difference in the tests as far as i could tell.

**Review by marandaneto — APPROVED** [review]

**Pull request merged**